### ISSUE TITLE
[nextercism] remove all exercise test versioning

### ISSUE THREAD

**Issue by robphoenix** [exercism/go#857]
With the new site we won't need to have the test versioning constants and tests.

~~@kytrinyx mentioned this in a comment, and I can't for the life of me remember where. If I find it I'll link to it here. I also don't remember why~~ :roll_eyes: 

~~This will need to be clarified by @kytrinyx in due course, and any related PR's merged into the nextercism branch rather than master.~~ (See comment below)

To be removed:

from test file
```
const targetTestVersion = 1

func TestTestVersion(t *testing.T) {
	if testVersion != targetTestVersion {
		t.Fatalf("Found testVersion = %v, want %v", testVersion, targetTestVersion)
	}
}
```
from example file & any stub
```
const testVersion = 1
```

**Comment by robphoenix** [comment]
Found Katrina's reference to removing test versioning: https://github.com/exercism/go/pull/807#issue-251726752

> when we launch the new v2 prototype publically, we will be able to get rid of the testVersions entirely, since we will always show the mentors the exact test suite that the user downloaded.

This is pretty clear I think, so I'm happy to go ahead with efforts to close this issue if there's no objections @tleen @leenipper @ferhatelmas

**Comment by robphoenix** [comment]
Closed by #861

**Comment by kytrinyx** [comment]
I'm so excited to see this improvement! I'll eventually open issues across the board for all the tracks, but it's so nice to see this go away.
